src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java

```diff
@@ -1260,6 +1269,30 @@ private void prepareSort(Query query, SearchRequestBuilder searchRequestBuilder,
 			}
 		}
 	}
+
+	private QueryRescorerBuilder getQueryRescorerBuilder(RescorerQuery rescorerQuery) {
+
+		QueryRescorerBuilder builder = new QueryRescorerBuilder(Objects.requireNonNull(getQuery(rescorerQuery.getQuery())));
```

Comment:
Instead of a `NullPointerException` we should throw an `IllegalArgumentException` here